pkgs/applications/science/math/weka/default.nix

```diff
@@ -19,18 +23,70 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [
     makeWrapper
     unzip
+    copyDesktopItems
+    imagemagick
   ];
 
   # The -Xmx1000M comes suggested from their download page:
   # https://www.cs.waikato.ac.nz/ml/weka/downloading.html
   installPhase = ''
+    runHook preInstall
+
     mkdir -pv $out/share/weka
+    mkdir -p $out/share/icons/hicolor
     cp -Rv * $out/share/weka
 
     makeWrapper ${jre}/bin/java $out/bin/weka \
       --add-flags "-Xmx1000M -jar $out/share/weka/weka.jar"
+
+    makeWrapper ${jre}/bin/java $out/bin/weka-java \
+      --add-flags "-Xmx1000M -cp $out/share/weka/weka.jar"
+
+    makeWrapper ${xdg-utils}/bin/xdg-open $out/bin/weka-doc \
```

Comment:
xdg-open will not work on darwin, consider gating it behind `isLinux` or find a darwin equivalent